index.js

```diff
@@ -133,6 +133,32 @@ function wrap (argv, env, workingDir) {
       options.envPairs.push((isWindows ? 'Path=' : 'PATH=') + workingDir)
     }
 
+    if (file === 'npm') {
```

Comment:
I'd be tempted to isolate this logic in a helper function, I find it's a good way to keep icky code like this isolated so that people can just concentrate on the inputs and outputs.